src/module/rules/rule-element/flat-modifier.ts

```diff
@@ -127,6 +127,7 @@ class FlatModifierRuleElement extends RuleElementPF2e {
                     critical: this.critical,
                     hideIfDisabled: this.hideIfDisabled,
                     source: this.item.uuid,
+                    unidentified: this.item.isOfType("effect") && this.item.system.unidentified,
```

Comment:
Rather than doing this, check if the item is an instanceof AbstractEffect and check for this.item.unidentified.

Reply:
very well